core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -5919,6 +5921,30 @@ private Object coerce(Type sourceType, Object value, Type targetType)
         }
     }
 
+    private static RefreshType getRefreshType(Query query)
```

Comment:
We should also have engine only tests too (apart from Iceberg tests). Especially since the logic here is not trivial and some coverage is missing. See `TestRefreshMaterializedView` for example. Maybe it can be extended or separate `BasePlanTest` could be added. Alternatively `TestMaterializedViews` could be extended. We could validate that correct `RefreshType` is used for different query shapes

Reply:
> We could validate that correct RefreshType is used for different query shapes

Addressed this in the `TestIncrementalRefreshVisitor` test